### DISCUSSION TITLE
One Command Multiple Events

### DISCUSSION THREAD

**Discussion by laalkep** [rasmus/EventFlow#920] [Q&A] [answered]
Hi, in the documentation at [docs.geteventflow.net](https://docs.geteventflow.net/DosAndDonts.html) there is an FAQ regarding emitting multiple events and being _careful_.

> **Subscribers and out of order events** - Be very careful if aggregates emit multiple events for a single command, subscribers will almost certainly receive these out of order.

That paragraph contains a link to [Out of order events](https://docs.geteventflow.net/Subscribers.html#out-of-order-event-subscribers) which explains a scenario where a command triggers 2 events and a subscriber of these events publishes another event.  And that those events will ultimately be consumed in a (_some might say_) unintuitive order.

I guess the _careful_ in the original statement relates to _being careful that you don't require or expect events to be in a specific order when emitting events from subscribers_.

1. Is my point about _careful_ correct?
2. Is the order of events only impacted when a subscriber publishes a command in response to an event?  For instance if I publish a command and it emits 4 events, are those four (if you were to exclude any events emitted by subscribers) guaranteed to be in correct order? 
3. If I am publishing a single command and emitting multiple events from that single command, are there any other (careful) things, to which, I need to be aware?

Kep

**Comment by rasmus** [comment, marked as answer]
Hi @kepboy you are correct. How the ordering plays out might be influenced how your domain is setup, so test it to be sure.
